internal/runners/config/meta.go

```diff
@@ -0,0 +1,23 @@
+package config
+
+import "github.com/ActiveState/cli/internal/constants"
+
+type event func() error
+
+type configType int
+
+const (
+	Int configType = iota
+	Bool
+)
+
+type configMeta struct {
+	allowedType configType
+	getEvent    event
+	setEvent    event
+}
+
+var meta = map[string]configMeta{
+	constants.SvcConfigPid:  {Int, nil, nil},
+	constants.SvcConfigPort: {Int, nil, nil},
+}
```

Comment:
nit: Ultimately, IIUC, this logic makes most sense in the internal/config package. It would need to be configured either kinda-globally in that config package, or structurally in the main package to be passed into the internal/config construction func. For this PR, it's not being used, and it looks like it can be moved easily enough, so no worries for now.

Reply:
If either you or @Naatan feels strongly about this being in the `internal/config` package I can move it. I didn't want to introduce the config meta to that package as I didn't want the `internal/config` package to have knowledge of specific config keys. If anything this could be its own package outside of both? But it felt better in the runner as that's where we need to apply rules to setting/getting config values.

Reply:
For now, I think it's best to land this as it is. If we do end up moving it into internal/config, we can work through how to ensure there's separation of concerns at that time.